### ISSUE TITLE
Cannot change local time, Ubuntu 16.04

### ISSUE THREAD

**Issue by badsector3** [Stromweld/timezone_iii#3]
Hi.

I use the cookbook to change the default timezone to 'Europe/London' but it does not change anything.
```ruby
role[testing]
name "test"
description "A role just for testing."
run_list "recipe[timezone_iii]"
        
default_attributes(
)

override_attributes(
    #timezone attributes
    "timezone_iii" => {
          "timezone" => "Europe/London",
          "use_symlink" => "true"
    }
)
```
but getting:
```bash
Starting Chef Client, version 12.19.36
resolving cookbooks for run list: ["timezone_iii"]
Synchronizing Cookbooks:
  - timezone_iii (1.0.3)
Installing Cookbook Gems:
Compiling Cookbooks...
Converging 4 resources
Recipe: timezone_iii::default
  * apt_package[tzdata] action install (up to date)
Recipe: timezone_iii::debian
  * template[/etc/timezone] action create
    - update content in file /etc/timezone from 2c4d4c to 188731
    --- /etc/timezone	2017-06-06 12:05:50.303619956 +0000
    +++ /etc/.chef-timezone20170606-7997-14fd1lu	2017-06-06 12:10:42.808840863 +0000
    @@ -1,2 +1,2 @@
    -Africa/Dakar
    +Europe/London
  * execute[dpkg-reconfigure-tzdata] action nothing (skipped due to action :nothing)
  * log[if-unexpected-timezone-change] action nothing (skipped due to action :nothing)
  * execute[dpkg-reconfigure-tzdata] action run
    [execute]
              Current default time zone: 'Africa/Dakar'
              Local time is now:      Tue Jun  6 12:10:42 GMT 2017.
              Universal Time is now:  Tue Jun  6 12:10:42 UTC 2017.

    - execute /usr/sbin/dpkg-reconfigure -f noninteractive tzdata
  * log[if-unexpected-timezone-change] action write

Running handlers:
Running handlers complete
Chef Client finished, 3/6 resources updated in 02 seconds

root@ip-10-0-30-18:/etc# timedatectl
      Local time: Tue 2017-06-06 12:10:48 GMT
  Universal time: Tue 2017-06-06 12:10:48 UTC
        RTC time: Tue 2017-06-06 12:10:48
       Time zone: Africa/Dakar (GMT, +0000)
 Network time on: yes
NTP synchronized: yes
 RTC in local TZ: no`
```
Its touch the timezone file but does not update it anyhow.

**Comment by Stromweld** [comment]
What OS and version are you running it on?

**Comment by badsector3** [comment]
"Ubuntu 16.04.2 LTS"

**Comment by Stromweld** [comment]
Ok yeah I don't work on Debian or ubuntu hardly ever, but someone else also hit me up with the issue and mentioned the old way was to set it in a file but now they are using the timedatectl command. I'll try to look at it today and see if I can get it working for you.

**Comment by badsector3** [comment]
Thanks. For now I wrote something simple:

Add it at the beginning of debian.rb
```ruby
if File.file?('/etc/localtime') then
  bash 'check-localtime' do
    user 'root'
    cwd '/tmp'
    code <<-EOH
    ls -la /etc/localtime >> /tmp/check.txt
    if grep node['timezone_iii']['timezone'] /tmp/check.txt ; then
      echo "Nothing to do!!!"
    else
      rm /etc/localtime
    fi
    rm /tmp/check.txt
    EOH
  end
end
```

It works as long as there is symlink between /usr/share/zoneinfo and /etc/localtime

**Comment by Stromweld** [comment]
cool thanks, I'll throw that in quick and cut a release as temp fix till better one can be made.

**Comment by QuingKhaos** [comment]
Somehow this script makes problems. And doesn't converge correctly and yeah, and it's not the chef way of doing things :smirk: 

In the first run it does this:

```
Recipe: timezone_iii::debian
  * template[/etc/timezone] action create
    - update content in file /etc/timezone from f0dcac to f7ffca
    --- /etc/timezone	2017-10-30 12:15:11.680000000 +0000
    +++ /etc/.chef-timezone20171030-11242-16qnk4k	2017-10-30 12:16:41.940000000 +0000
    @@ -1,2 +1,2 @@
    -Etc/UTC
    +Europe/Vienna
  * execute[dpkg-reconfigure-tzdata] action nothing (skipped due to action :nothing)
  * log[if-unexpected-timezone-change] action nothing (skipped due to action :nothing)
Recipe: timezone_iii::debian
  * execute[dpkg-reconfigure-tzdata] action run
    - execute /usr/sbin/dpkg-reconfigure -f noninteractive tzdata
```

I go into the server, the timezone is correct. In the next run, the following is happening:

```
Recipe: timezone_iii::debian
  * bash[check-localtime] action run
    - execute "bash"  "/tmp/chef-script20171030-16779-1ncbdr9"
  * template[/etc/timezone] action create (up to date)
  * execute[dpkg-reconfigure-tzdata] action nothing (skipped due to action :nothing)
  * log[if-unexpected-timezone-change] action nothing (skipped due to action :nothing)
```

You see, the bash script is now executed and somehow removes the `/etc/localtime` symlink, altough it shouldn't(?)

```# ls -la /etc/localtime
ls: cannot access '/etc/localtime': No such file or directory
```

Which means my timezone is now back to UTC. In the third run it does converge nothing :sob: 

```
Recipe: timezone_iii::debian
  * template[/etc/timezone] action create (up to date)
  * execute[dpkg-reconfigure-tzdata] action nothing (skipped due to action :nothing)
  * log[if-unexpected-timezone-change] action nothing (skipped due to action :nothing)
```

**Comment by QuingKhaos** [comment]
TL;DR: And very hard to see, the `node['timezone_iii']['timezone']` in the script is not a variable interpolation. So it's grepping for `node['timezone_iii']['timezone']` in the `check.txt`, not for `Europe/Vienna` in my case. Patch incoming.

**Comment by QuingKhaos** [comment]
PR pushed. Although it works now, it should be improved. As said, the bash mash is not the chef way ;) And thus it always shows up as changed resource in the chef-client run on every run. The script itself is somehow idempotent, but not idempotent in the converge of chef:

```
Recipe: timezone_iii::debian
  * bash[check-localtime] action run
    - execute "bash"  "/tmp/chef-script20171030-16916-vc1t8p"
  * template[/etc/timezone] action create (up to date)
  * execute[dpkg-reconfigure-tzdata] action nothing (skipped due to action :nothing)
  * log[if-unexpected-timezone-change] action nothing (skipped due to action :nothing)
```

**Comment by limitusus** [comment]
How about replacing `/etc/timezone` with `template` resource, with notifying `:delete` `/etc/localtime` symlink and `:run` `dpkg-reconfigure` ?
It can avoid from using the bash mash.

I just faced the same issue and see it solved like the following recipe
```ruby
LOCALTIME_FILE = '/etc/localtime'.freeze

template "/etc/timezone" do
  source "timezone.conf.erb"
  owner 'root'
  group 'root'
  mode 0644
  notifies :delete, "file[#{LOCALTIME_FILE}]"
  notifies :run, 'bash[dpkg-reconfigure tzdata]'
end

file LOCALTIME_FILE do
  action :nothing
end

bash 'dpkg-reconfigure tzdata' do
  user 'root'
  code "/usr/sbin/dpkg-reconfigure -f noninteractive tzdata"
  action :nothing
end
```

**Comment by Stromweld** [comment]
merged commit should fix issue. Feel free to clean up code and submit a new PR if like to make it more chef like. Personally I haven't worked with debian or ubuntu for years now that this code is from my copying of the timezone_ii cookbook.
